### ISSUE TITLE
Refactor code according to your proposed style

### ISSUE THREAD

**Issue by dgentil1** [dgentil1/Katrina#3]
Please do this changes before Sunday mid day (US time) so that I can review them before our meeting.

**Comment by dgentil1** [comment]
As spoken in our Skype meeting, this was a good job. However, before we close this issue let's make sure that we agree on  what we want. In particular, are we sure we want to keep the silent thing? As I told you the other day, I am not sure we care to have a nice log file given that our final results will live in the following folders, which are clear enough to understand what they contain: 
tables, figures, draft, slides and maps

If it turns out that we decide to keep this 100 lines of code per do file  I want to make sure that I understand why. Could you please explain me why do you think this is beneficial for us? 

Summarizing:

**Cons**
- do files are less readable and much longer

**Pros**
- still unclear

**Comment by marinabosque** [comment]
I have already erase all the silent thing. As you proposed I have just added in the main dofile a global variable called qui, in case we need this nice output.

So:
If we want to see the output --> Main dofile ---> global qui ""
If we don't want to see the output --> Main dofile ---> global qui "qui"

**Comment by dgentil1** [comment]
Amazing news Mar, this is definitely an improvement! I think we can lose this issue then if you all agree

**Comment by danielealimonti** [comment]
Let's close this issue! I agree on Marina's modifications. Now we get a more readable dofile and simultaneously a nice stata output.

**Comment by dgentil1** [comment]
Great! I will close this task soon. Just some more questions and comments: 

1. What you guys called workdata is what I was calling derived right? 
2. I really like our new workflow, but let's try to keep the code evolution organized. Given that we all agree in the code that was developed on the branch. Please merge it back into the master so that you practice on how to do this. Also, remember that for avoiding code conflicts each time that we want to make a modification for a different reason we should do it on a branch named correspondingly and once we had someone reviewing each others work we should merge it back into master. Like this we keep it organized if we want to make crazy tests in the code at the same time that we keep the collaboration organised and efficient by reviewing each others stuff. 
3. Once the code is merged into master I will download it and make sure that it runs on my side. Once it does we can close this issue. 

Thanks!

**Comment by marinabosque** [comment]
1. I think so. @danielealimonti will tell you.
2. As soon as me and Daniele are together we will do it.
3. Ok

Thanks to you!

**Comment by danielealimonti** [comment]
1) Yes, it is. In particular, workdata contains another folder called work_asec (for the moment). Same reasoning for rawdata. Following this scheme, we will be able to keep organised our raw data and derived data for different sources (monthly data/asec/nber).

2) We will do it this morning.

3) We will notify you so you will download them.

**Comment by dgentil1** [comment]
My answers:

1. OK. Then, the workflow is that datafix.do takes the data from rawdata precleans it and does some stuff and then saves it to workdata? If that is how it goes I agree. 

2. Great. Per our recent conversations the to do list [here](https://github.com/dgentil1/Katrina/blob/restyle/code/readme.txt) should be deleted. We don't want comments, nor display command that make our do files tedious and rearranging and cleaning the do files will be the very last step once it is clear what makes it to the paper. Once we arrive to that stage we will just preserve the code that produces strictly what the paper shows. 

3. Let me know when you are ready and I'll do the testing :D

**Comment by marinabosque** [comment]
1. Correcto!

2. Erased!

3. Ready!

4. IMPORTANT!! In the Endogeneity.do, it is not possible to run:


************
  ***** Merge
  
	${qui} use "${work_asec}lot_evac_list_to_match.dta", clear

	${qui} merge 1:1 metcode2 using "${temp}distance_matrix.dta", nogen
	${qui} merge 1:1 metcode2 using "${work_asec}lot_evac_list.dta", nogen

	${qui} export delimited using "${layers}distance_matrix_matched.csv", replace

	label var distance "Distance to New Orleans"
	label var unem_1 "Average unemployment rate, last year"
	label var unem_5 "Average unemployment rate, last 5 years"
	label var lr_w_wage_1 "Average log-weekly wage, last year"
	label var lr_w_wage_5 "Average log-weekly wage, last 5 years"
	label var hourwage_1 "Average hourly wage, last year"
	label var hourwage_5 "Average hourly wage, last 5 years"
	label var share_evac "Evacuees share"
	label var treat "Treatment"

	eststo clear

	${qui} eststo: reg share_evac distance unem_1 lr_w_wage_1
	${qui} eststo: reg share_evac distance unem_5 lr_w_wage_5

	${qui} esttab using "${tables}treatment_assignment.tex", r2 se nocons ///
		   label compress replace title(Treatment assignment)
************

We don't have the dataset: 'lot_evac_list_to_match.dta'!!!!!!

5. If you run from the main.do you will run everything at once, EXCEPT the PLACEBOS that are "disactivated". If you wanna activate them, go to --> Synthetic_main.do and erase the start* that is blocking it!!

6. I have just tried all the dofies (except the endogeneity and placebos) and they work in my pc!! I don't know why, but when I run the S_SCM.do, Stata gives me this error "system limit exceeded - see manual".  I think I am the only one who is gonna have this problem, but if you can check it and tell me I would appreciate it!

**Comment by dgentil1** [comment]
Great work guys! I will test everything and I will update the preclean so that endogeneity works as well.

**Comment by danielealimonti** [comment]
Can I mark this issue as solved? Did you test the code?

**Comment by dgentil1** [comment]
Guys I am sorry but I didn't had time, I'm inundated of work until tomorrow. I will update you as soon as I can. For now, I've added your code to trunk, so whatever tasks you are doing you can create a branch and add the code there and we will merge it to trunk later! Thanks

**Comment by dgentil1** [comment]
Guys, I have been reviewing the code and testing it widely. Some comments:

1. having ${qui} on each line makes the code much harder to read on each pass and we are not gaining anything else but a clean log file which we don't need in the first place. Also, coding adding that line every time is not intuitive at all, so after coding it is not good use of our time to have to add that phrase to each line before committing. Please let's remove those. 
2.  What do we gain by using:
	`cap saveold "${temp}CPSASEC.dta", v(12) replace`
Instead of just saving in the current version?
3. In line 76 of D_idkatrina there is a comment asking for an explanation. Let me know what can I explain, but please let's avoid communicating through the do file as it is not very efficient and makes the do files harder to read and maintain (as we then have to remember to delete those comments). 

I am almost ready for committing my work in several fronts so once you are ready with these changes I will do! :D 

Let's target to have this changes plus my commits for assigning treatment, adding lot_evac_list_to_match.dta and the basic preclean for the monthly data by tomorrow night so that we can speak for around half an hour on Sunday. I will also take care of moving the computation of the tabstats to a different do file as it was before. I don't think it makes sense to have that in the middle of the script where we identify treated units. I believe that ideally we want to compute all the tabstats that we show in a single script, however If you think that there is a reason for this not being correct let me know and we can discuss it.

**Comment by marinabosque** [comment]
1. Removed

2. Daniele has Stata14 and I have Stata12. We cannot exchange datasets because I cannot open a Stata14 dataset in my Stata version.

3. We ask you today. I removed the comment from the dofile.

Talk to you tonight guys :)

**Comment by dgentil1** [comment]
Gracias Mar :D

1. Buena onda
2. Understood, then makes sense let's keep it like that. 
3. Perfect, let me know I'll be available all day today!

**Comment by dgentil1** [comment]
Guys, I've committed a version in which:

1. we use relative paths so it should work for both mac and windows (Please Mar confirm whenever you can!) You will have to close the repository into a new location and in that location you will have to create at the same level as GIS and code the following directories (I will send you a picture of how it should look!):
- temp
- derived_asec
- figures
- tables
- raw (please add the data here)
 
2. Fixed the a couple things that were not running: 

- synth
- endogeneity
- descriptive statistics

3. added a variable called treat_expanded in which I flag observations coming from 9 treated cities (including our previous 3)

4. I've added a .gitignore file so that once you set up a folder containing the whole repository git will be smart and not commit the derived, temp, tables, figures and raw. 

Once you confirm we are clear to close here :D

**Comment by dgentil1** [comment]
This is done!
